src/main/resources/graphql/content.graphqls

```diff
@@ -0,0 +1,118 @@
+interface Content {
+    """
+    ID of the content
+    """
+    id: UUID!
+
+    """
+    Metadata of the content
+    """
+    metadata: ContentMetadata!
+}
+
+type ContentMetadata {
+    """
+    Name of the content
+    """
+    name: String!
+    """
+    Content type
+    """
+    type: ContentType!
+    """
+    Number of reward points a student receives for completing this content
+    """
+    rewardPoints: Int!
+    """
+    ID of the chapter this content is associated with
+    """
+    chapterId: UUID!
+    """
+    TagNames this content is tagged with
+    """
+    tagNames: [String!]!
+}
+
+type MediaContent implements Content {
+    # Inherited fields
+    """
+    ID of the content
+    """
+    id: UUID!
+
+    """
+    Metadata of the content
+    """
+    metadata: ContentMetadata!
+}
+
+interface Assessment implements Content {
+
+    """
+    Assessment metadata
+    """
+    assessmentMetadata: AssessmentMetadata!
+
+    # inherited fields
+    """
+    ID of the content
+    """
+    id: UUID!
+    """
+    Metadata of the content
+    """
+    metadata: ContentMetadata!
+
+}
+
+type AssessmentMetadata {
+    """
+    Number of skill points a student receives for completing this content
+    """
+    skillPoints: Int! # TODO skill concept done in the future
+    """
+    Type of the assessment
+    """
+    type: AssessmentType!
+}
+
+"""
+A set of flashcards, flashcard related fields are stored in the flashcard service.
+"""
+type FlashcardSetAssessment implements Assessment & Content {
+    # inherited fields
+    """
+    Assessment metadata
+    """
+    assessmentMetadata: AssessmentMetadata!
+    """
+    ID of the content
+    """
+    id: UUID!
+    """
+    Metadata of the content
+    """
+    metadata: ContentMetadata!
+}
+
+# add more types here
+
+"""
+Type of the assessment
+"""
+enum AssessmentType {
```

Comment:
Is this really what Assessment type is supposed to be? And not just something like "Quiz", "Flashcard" etc.?

Reply:
no, check domain model